### DISCUSSION TITLE
How to make the predicted value non trivial？

### DISCUSSION THREAD

**Discussion by chuanhon** [lululxvi/deepxde#1252] [Q&A] [answered]
The question is -u''=λu,  λ is unkown, and 0<x<pi, and u(0)=u(pi)=0.
Code as follow,
import deepxde as dde
import matplotlib.pyplot as plt
import numpy as np

def pde(x, y):
    dy_xx = dde.grad.hessian(y, x)
    return -dy_xx - y

def boundary(x, _):
    return np.isclose(x[0], 0) or np.isclose(x[0], np.pi)

def func(x):
    return np.sin(x)

test_train=np.random.uniform(0,np.pi,50)
TestTrain=test_train.reshape(50,1)

geom = dde.geometry.Interval(0, np.pi)
bc = dde.icbc.DirichletBC(geom, func, boundary)
data = dde.data.PDE(geom, pde, bc, 50, 2, anchors=TestTrain, solution=func, num_test=100)

layer_size = [1] + [50] * 3 + [1]
activation = "tanh"
initializer = "Glorot uniform"
net = dde.nn.FNN(layer_size, activation, initializer)

model = dde.Model(data, net)
model.compile("adam", lr=0.001, metrics=["l2 relative error"])
losshistory, train_state = model.train(iterations=10000)

dde.saveplot(losshistory, train_state, issave=True, isplot=True)

![myplot](https://user-images.githubusercontent.com/81275002/232449618-db027ef5-2e87-4181-b1cd-a69803725ebe.png)
![myplot1](https://user-images.githubusercontent.com/81275002/232449626-faf3aad7-31c5-4870-b033-76380234e47d.png)
see,it predicted 0

**Comment by haison19952013** [comment]
You can refer to this one: https://github.com/lululxvi/deepxde/issues/356#issue-971114756...

**Reply by chuanhon** [reply]
thanks.It do help

**Comment by pescap** [comment, marked as answer]
Hi,
You are solving:
```
-dy_xx - y = 0 
y(0) = y(\pi) = 0
```
This problem has no source (or forcing) term and boundary data are `0`. The solution to your problem as it is posed is `y=0`.
